### PR TITLE
Nes 906 group name

### PR DESCRIPTION
**Pull request by pc-m**
Depends-On: https://github.com/wazo-platform/xivo-dao/pull/121

### CODE REVIEW

File: integration_tests/suite/base/test_event_handlers.py [hunk 1]
```diff
@@ -35,4 +36,9 @@ def templates_created():
             ),
         )
 
+    def slut_created():
+        # There's no API to check the tenant slug. It is part of any new group name
+        group = confd.groups.create({'label': 'ignore'}, wazo_tenant=CREATED_TENANT)
+        assert_that(group, has_entries(name=starts_with('grp-myslug-')))
+
     until.assert_(templates_created, tries=5)
```
Comment (fblackburn1) [inline integration_tests/suite/base/test_event_handlers.py:44]:
```suggestion
    def slut_created():
        # There's no API to check the tenant slug. It is part of any new group name
        group = confd.groups.create({'label': 'ignore'}, wazo_tenant=CREATED_TENANT)
        assert_that(group, has_entries(name=starts_with('grp-myslug-')))

    until.assert_(templates_created, tries=5)
    until.assert_(slug_created, tries=5)
```

File: wazo_confd/plugins/group/validator.py [hunk 2]
```diff
@@ -18,20 +14,12 @@ class GroupValidator(Validator):
     def validate(self, group):
         self.validate_unique_name_through_queue(group)
 
+    # TODO(pc-m): remove this validation once queues get there name generated
     def validate_unique_name_through_queue(self, group):
         queue = queue_dao.find_by(name=group.name)
         if queue is not None:
             raise errors.resource_exists('Queue', name=group.name)
```
Comment (fblackburn1) [inline wazo_confd/plugins/group/validator.py:21]: Can we remove this?

File: wazo_confd/sync_db.py [hunk 3]
```diff
@@ -87,6 +92,13 @@ def main():
             tenant = tenant_dao.find_or_create_tenant(tenant_uuid)
             default_sip_template_service.generate_sip_templates(tenant)
 
+        for tenant_uuid in confd_tenant_without_slugs:
+            slug = auth_tenant_slugs.get(tenant_uuid)
+            if not slug:
+                continue
+            tenant = tenant_dao.find_or_create_tenant(tenant_uuid)
+            tenant.slug = slug
```
Comment (fblackburn1) [inline wazo_confd/sync_db.py:100]: by good practice we should flush session after this operation (with flush directly or wrap it in `tenant_dao.update(tenant)`)

File: integration_tests/suite/base/test_event_handlers.py (hunk 1 above)
Comment (pc-m) [reply integration_tests/suite/base/test_event_handlers.py:44]: Done, I also fixed the `slug_created` typo

File: wazo_confd/plugins/group/validator.py (hunk 2 above)
Comment (pc-m) [reply wazo_confd/plugins/group/validator.py:21]: I'd rather wait until we do the same thing with queues and avoid someone messing around with queue name and end up doing something that breaks everything and then open a bug.

File: wazo_confd/helpers/resource.py [hunk 4]
```diff
@@ -39,3 +39,6 @@ def delete(self, resource):
         self.validator.validate_delete(resource)
         self.dao.delete(resource)
         self.notifier.deleted(resource)
+
+    def update_form(self, form):
+        return form
```
Comment (fblackburn1) [inline wazo_confd/helpers/resource.py:44]: This layer should receive only object (sqlalchemy) or id. The form transformation is done in the restful layer

**Review by fblackburn1 — CHANGES_REQUESTED** [review]

File: wazo_confd/plugins/group/validator.py (hunk 2 above)
Comment (fblackburn1) [reply wazo_confd/plugins/group/validator.py:21]: Ok :+1:

File: wazo_confd/sync_db.py (hunk 3 above)
Comment (pc-m) [reply wazo_confd/sync_db.py:100]: which does not exist :sweat_smile:

**Comment by fblackburn1** [comment]
recheck

File: wazo_confd/sync_db.py (hunk 3 above)
Comment (fblackburn1) [reply wazo_confd/sync_db.py:100]: Hahaha

**Review by fblackburn1 — APPROVED** [review]

**Review by fblackburn1 — DISMISSED** [review]

**Pull request merged**